### PR TITLE
Update Build.cmd to reflect new VS and options

### PR DESCRIPTION
**Pull request by davkean**
- Update build.cmd to reflect VS 2017 changes
- Write build fail and success in color
- Update build.cmd help to reflect changes

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/project-system

File: build.cmd [hunk 1]
```diff
@@ -22,49 +21,54 @@ if /I "%1" == "/no-multi-proc" set MSBuildAdditionalArguments=&&shift&& goto :Pa
 call :Usage && exit /b 1
 :DoneParsing
 
-if not exist "%DeveloperCommandPrompt%" (
-  echo In order to build this repository, you need Visual Studio "15" Preview installed.
+if not exist "%VSINSTALLDIR%" (
```
Comment (mavasani) [inline build.cmd]: Will this break microbuild @basoundr

Comment (davkean) [inline build.cmd]: I don't know - how does it work today? VS doesn't set any variables to find the developer command prompt.

Comment (basoundr) [inline build.cmd]: The variable needs to be added [here](https://devdiv.visualstudio.com/DevDiv/_build/explorer?_a=variables&definitionId=1987)

Comment (davkean) [inline build.cmd]: Looks like VS150COMNTOOLS is being set in jenkins and VSO - I'll respect that instead.

File: build.cmd [hunk 2]
```diff
@@ -22,49 +21,54 @@ if /I "%1" == "/no-multi-proc" set MSBuildAdditionalArguments=&&shift&& goto :Pa
...
   exit /b 1
 )
 
 if not exist "%VSSDK150Install%" (
-  echo In order to build this repository, you need to modify your Visual Studio installation to include "Visual Studio Extensibility Tools".
+  echo To build this repository, you need to modify your Visual Studio installation to include the "Visual Studio extension development" workload.
   exit /b 1
 )
 
-call "%DeveloperCommandPrompt%" || goto :BuildFailed
```
Comment (mavasani) [inline build.cmd]: Again, are we sure that not running VsDevCmd.bat will not break microbuild? Tagging @basoundr

Comment (davkean) [inline build.cmd]: Yes, it looks like it breaks both MicroBuild and Jenkins, changing this.

**Pull request merged**